spec/controllers/settings/follower_domains_controller_spec.rb

```diff
@@ -5,32 +5,81 @@
 
   let(:user) { Fabricate(:user) }
 
-  before do
-    sign_in user, scope: :user
+  shared_examples 'authenticate user' do
+    it 'redirects when not signed in' do
+      is_expected.to redirect_to '/auth/sign_in'
+    end
   end
 
   describe 'GET #show' do
+    subject { get :show, params: { page: 2 } }
+
+    it 'assigns @account' do
+      sign_in user, scope: :user
+      subject
+      expect(assigns(:account)).to eq user.account
+    end
+
+    it 'assigns @domains' do
+      followers = 11.times.map do |index|
```

Comment:
Can we move the huge string of methods in the controller into some scopes and class methods instead? That might make it easier to have a more thorough spec for the query improvement (in a model spec) while keeping the controller spec more focused, and the controller code more readable.